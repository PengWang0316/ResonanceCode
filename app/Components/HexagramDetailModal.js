import React from 'react';
import PropTypes from 'prop-types';

import ImageDescription from './ImageDescription';
// import AssociatedHexagrams from './AssociatedHexagrams';
import GroupHexagramTable from './GroupHexagramTable';
import BigramClockBig from './BigramClockBig';
import BigramBlockBig from './BigramBlockBig';
import ChangingLines from './ChangingLines';
import styles from '../styles/HexagramDetailModal.module.css';

const getGroupHexagramObject = (name, number, imageArr, rcName) => ({
  name, number, imageArr, rcName
});

const getQuartetHexagramArray = hexagram => [
  getGroupHexagramObject('Potentiation', hexagram.potentiation_hexagram_number, hexagram.potentiation_hexagram, hexagram.potentiation_hexagram_code),
  getGroupHexagramObject('Growth', hexagram.growth_hexagram_number, hexagram.growth_hexagram, hexagram.growth_hexagram_code),
  getGroupHexagramObject('Maturation', hexagram.maturation_hexagram_number, hexagram.maturation_hexagram, hexagram.maturation_hexagram_code),
  getGroupHexagramObject('Re-Sourcing', hexagram.resourcing_hexagram_number, hexagram.resourcing_hexagram, hexagram.resourcing_hexagram_code)
];

const getAssociateHexagramArray = hexagram => [
  getGroupHexagramObject('Complementary', hexagram.complementary_hexagram_number, hexagram.complementary_hexagram, hexagram.complementary_hexagram_code),
  getGroupHexagramObject('Reverse', hexagram.reverse_hexagram_number, hexagram.reverse_hexagram, hexagram.reverse_hexagram_code),
  getGroupHexagramObject('Hidden Influence', hexagram.hidden_influence_hexagram_number, hexagram.hidden_influence_hexagram, hexagram.hidden_influence_hexagram_code)
];

const HexagramDetailModal = ({ hexagram, handleHexagramClick, hexagramsImgArrMap }) => (
  <div className="modal fade bd-example-modal-lg" id="hexagramDetailModal" tabIndex="-1" role="dialog" aria-labelledby="hexagramDetailModalLabel" aria-hidden="true">
    <div className="modal-dialog modal-lg" role="document">
      <div className="modal-content">
        <div className="modal-header">
          <h5 className="modal-title" id="hexagramDetailModalLabel">&nbsp;</h5>
          <button type="button" className="close" data-dismiss="modal" aria-label="Close">
            <span aria-hidden="true">&times;</span>
          </button>
        </div>
        {hexagram &&
          <div className="modal-body">
            <ImageDescription
              imageInfo={hexagram}
              imageNumber={hexagram.img_arr}
              isFirstImage
              isBlack
            />
            <div className="mt-4 d-flex flex-wrap justify-content-center">
              <BigramClockBig lineText="Resonance Line 2/5" position={hexagram.line_25} />
              <BigramClockBig lineText="Agency/Process Line 1/4" position={hexagram.line_14} />
              <BigramClockBig lineText="Agent/Identity Line 3/6" position={hexagram.line_36} />
            </div>
            <div className="pl-4 mt-5 d-flex flex-wrap justify-content-around">
              <BigramBlockBig lineText="Particle Bigram: Line 1 and 3" position={hexagram.line_13} isLine13 />
              <BigramBlockBig lineText="Wave Bigram: Line 4 and 6" position={hexagram.line_46} />
            </div>
            <div className={`mt-4 ${styles.imagePoetryDiv}`}>
              {hexagram.image && <img src={hexagram.image} alt="hexagram" className={`img-fluid float-sm-left ${styles.image}`} />}
              <blockquote
                className={styles.poetry}
                style={hexagram.poetry_font_size ? { fontSize: `${hexagram.poetry_font_size}px` } : {}}
              >
                {hexagram.poetry}
              </blockquote>
            </div>
            <div className={`mt-4 ${styles.overViewDiv}`}>Overview:</div>
            <div className={styles.preLineWhiteSpace}>{hexagram.overview}</div>
            <div className="mt-4 font-weight-bold">Question:</div>
            <div className={styles.preLineWhiteSpace}>{hexagram.question}</div>
            <div className="mt-4 font-weight-bold">Analysis:</div>
            <div className={`mb-4 ${styles.preLineWhiteSpace}`}>{hexagram.analysis}</div>
            <ChangingLines hexagram={hexagram} hexagramsImgArrMap={hexagramsImgArrMap} />
            <GroupHexagramTable
              hexagramArray={getAssociateHexagramArray(hexagram)}
              handleHexagramClick={handleHexagramClick}
              tableTitle="Associated Hexagrams"
            />
            <GroupHexagramTable
              hexagramArray={getQuartetHexagramArray(hexagram)}
              handleHexagramClick={handleHexagramClick}
              tableTitle="Resonance Quartet Hexagrams"
            />
          </div>
        }
        <div className="modal-footer">
          <button type="button" className="btn btn-secondary" data-dismiss="modal">Close</button>
        </div>
      </div>
    </div>
  </div>
);
HexagramDetailModal.propTypes = {
  hexagram: PropTypes.object,
  handleHexagramClick: PropTypes.func.isRequired,
  hexagramsImgArrMap: PropTypes.object.isRequired
};
HexagramDetailModal.defaultProps = {
  hexagram: null
};
export default HexagramDetailModal;
