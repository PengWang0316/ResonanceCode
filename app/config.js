export const JWT_MESSAGE = 'KairoScope_JWT';
export const ADMINISTRATOR_ROLE = 1;
// export const BASE_URL = 'https://kairoscope-backend.glitch.me'; // Development server
// export const BASE_URL = 'http://localhost:8080'; // Development server
export const BASE_URL = 'https://kairoscope.resonancepath.com:8080'; // Production server
export const AWS_READING_URL = 'https://b8bpiar0bh.execute-api.us-west-2.amazonaws.com/dev';
export const AWS_JOURNAL_URL = 'https://xbfdcju8sf.execute-api.us-west-2.amazonaws.com/dev';
export const AWS_HEXAGRAM_URL = 'https://3i0e59njre.execute-api.us-west-2.amazonaws.com/dev';
export const AWS_USER_URL = 'https://9p4s43slt2.execute-api.us-west-2.amazonaws.com/dev';
export const NUMBER_OF_READING_PER_PAGE_RECENT_READINGS = 5; // How many reading will show on the recent readings page.
export const NUMBER_OF_USER_PER_PAGE = 30; // How many user will be showed in the all user list.
export const NUMBER_OF_READING_PER_PAGE = 10; // How many reading will be showed in one page.
export const NUMBER_PER_PAGE_JOURNAL = 5; // How many journal will be showed in all journal page.
export const CLOUDINARY_UPLOAD_PRESET = 'KairoScopeWebApp'; // The preset of Cloudinary.
export const TOTAL_NUMBER_HEXAGRAM = 64; // The total amount Hexagram.
