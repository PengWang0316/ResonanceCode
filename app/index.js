import React from "react";
import ReactDOM from "react-dom";
import App from "./components/App";
import "./index.css";
// import jQuery from "jQuery";
// require('bootstrap');



ReactDOM.render(
  <App />,
  document.getElementById("app")
);
