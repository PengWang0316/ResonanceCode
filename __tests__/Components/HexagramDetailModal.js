import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { TOTAL_NUMBER_HEXAGRAM } from '../../app/config';
import { HexagramDetailModal } from '../../app/Components/HexagramDetailModal';

jest.mock('../../app/Components/ChangingLines', () => 'ChangingLines');
jest.mock('../../app/Components/ImageDescription', () => 'ImageDescription');
jest.mock('../../app/Components/GroupHexagramTable', () => 'GroupHexagramTable');
jest.mock('../../app/Components/BigramClockBig', () => 'BigramClockBig');
jest.mock('../../app/Components/BigramBlockBig', () => 'BigramBlockBig');
jest.mock('../../app/Components/Bigram', () => 'Bigram');
jest.mock('../../app/Components/BigramExample', () => 'BigramExample');


describe('HexagramDetailModal test', () => {
  const defaultProps = {
    hexagrams: [
      {
        _id: 'id2',
        number: 2
      },
      {
        _id: 'id',
        number: 1,
        img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
        resonance_code_name: 'resonance_code_name',
        wilhelm_huang_hintley_name: 'wilhelm_huang_hintley_name',
        image: 'image url',
        poetry: 'poetry content',
        overview: 'overview',
        analysis: 'analysis',
        question: 'queston',
        line_13: 0,
        line_25: 0,
        line_46: 0,
        line_14: 0,
        line_36: 0,
        changing_lines: {
          line1: {
            img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
            line_13: [1, 1],
            line_14: [1, 1]
          },
          line2: {
            img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
            line_25: [1, 1]
          },
          line3: {
            img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
            line_13: [1, 1],
            line_36: [1, 1]
          },
          line4: {
            img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
            line_46: [1, 1],
            line_14: [1, 1]
          },
          line5: {
            img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
            line_25: [1, 1]
          },
          line6: {
            img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
            line_46: [1, 1],
            line_36: [1, 1]
          }
        }
      }
    ],
    hexagramArr: '',
    hexagramsImgArrMap: {
      '7,9-7,9-7,9-7,9-7,9-7,9': { _id: 'id', number: 'number', resonance_code_name: 'code name' }
    },
    handleHexagramClick: jest.fn(),
    fetchHexagrams: jest.fn(),
    clearHexagrams: jest.fn()
  };
  const getShallowComponent = (props = defaultProps) => shallow(<HexagramDetailModal {...props} />);

  test('componentWillMount no hexagrams in props', () => {
    const mockFetchHexagrams = jest.fn();
    const mockclearHexagrams = jest.fn();
    getShallowComponent({
      hexagrams: [],
      fetchHexagrams: mockFetchHexagrams,
      clearHexagrams: mockclearHexagrams
    });
    expect(mockFetchHexagrams).toHaveBeenCalledTimes(1);
    expect(mockclearHexagrams).toHaveBeenCalledTimes(1);
  });

  test('componentWillMount has hexagrams in props', () => {
    const mockFetchHexagrams = jest.fn();
    const mockclearHexagrams = jest.fn();
    const hexagramArray = new Array(TOTAL_NUMBER_HEXAGRAM);
    hexagramArray[0] = { img_arr: '79', _id: 1 };
    const component = getShallowComponent({
      hexagrams: hexagramArray,
      fetchHexagrams: mockFetchHexagrams,
      clearHexagrams: mockclearHexagrams
    });
    expect(mockFetchHexagrams).toHaveBeenCalledTimes(0);
    expect(mockclearHexagrams).toHaveBeenCalledTimes(0);
    // expect(component.instance().hexagramsImgArrMap['79']).not.toBeUndefined();
  });

  test('componentWillReceiveProps', () => {
    const hexagramArray = new Array(TOTAL_NUMBER_HEXAGRAM);
    hexagramArray[0] = { img_arr: '68', _id: 1 };
    const mockFetchHexagrams = jest.fn().mockReturnValue(hexagramArray);
    const mockclearHexagrams = jest.fn();
    const component = getShallowComponent({
      hexagrams: [],
      fetchHexagrams: mockFetchHexagrams,
      clearHexagrams: mockclearHexagrams
    });
    expect(mockFetchHexagrams).toHaveBeenCalledTimes(1);
    expect(mockclearHexagrams).toHaveBeenCalledTimes(1);
    expect(component.instance().hexagramsImgArrMap).toBeUndefined();

    component.setProps({ hexagrams: hexagramArray });
    expect(component.instance().hexagramsImgArrMap['68']).not.toBeUndefined();
  });

  test('handleAssociatedHexagramClick', () => {
    const component = getShallowComponent();
    // Prepare the this.hexagrams data for the test.
    // component.find('tr').at(1).simulate('click', { target: { parentNode: { nodeName: 'TR', id: '1', getAttribute: jest.fn().mockReturnValue('1') } } });
    component.setState({
      hexagram: {
        _id: 'newId',
        number: 1,
        img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
        resonance_code_name: 'resonance_code_name',
        wilhelm_huang_hintley_name: 'wilhelm_huang_hintley_name',
        image: 'image url',
        poetry: 'poetry content',
        poetry_font_size: 18,
        overview: 'overview',
        analysis: 'analysis',
        question: 'queston',
        line_13: 0,
        line_25: 0,
        line_46: 0,
        line_14: 0,
        line_36: 0,
        changing_lines: {
          line1: {
            img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
            line_13: [1, 1],
            line_14: [1, 1]
          },
          line2: {
            img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
            line_25: [1, 1]
          },
          line3: {
            img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
            line_13: [1, 1],
            line_36: [1, 1]
          },
          line4: {
            img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
            line_46: [1, 1],
            line_14: [1, 1]
          },
          line5: {
            img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
            line_25: [1, 1]
          },
          line6: {
            img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
            line_46: [1, 1],
            line_36: [1, 1]
          }
        }
      }
    });
    // HexagramListContainer.getHexagramBaseOnTarget = jest.fn().mockReturnValue({ id: 'newId' });
    component.find('GroupHexagramTable').at(0).prop('handleHexagramClick')({ target: { parentNode: { nodeName: 'TR', number: '2', getAttribute: jest.fn().mockReturnValue('2') } } });
    expect(component.instance().hexagramNumbersMap[1]._id).toBe('id');
    expect(component.instance().state.hexagram._id).toBe('id2');
    // expect(component.state('hexagram')).toBe(defaultProps.hexagrams[2]);
  });

  test('handleBigramClick', () => {
    const component = getShallowComponent();
    expect(component.instance().state.isShowExample).toBe(false);
    component.instance().handleBigramClick();
    expect(component.instance().state.isShowExample).toBe(true);
    component.instance().handleBigramClick();
    expect(component.instance().state.isShowExample).toBe(false);
  });

  test('has BigramExample', () => {
    const component = getShallowComponent();
    component.setState({ hexagram: true });
    expect(component.find('BigramExample').length).toBe(0);
    component.setState({ isShowExample: true });
    expect(component.find('ChangingLines').length).toBe(1);
  });

  /* Using this method to add snapshot will cause a "unique key" warnning.
     More investigation needed.
  test('HexagramDetailModal snapshot', () => {
    const component = getShallowComponent();
    component.instance().hexagramsImgArrMap = {
      '7,9-7,9-7,9-7,9-7,9-7,9': {
        _id: 'id',
        number: 1,
        img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
        resonance_code_name: 'resonance_code_name',
        wilhelm_huang_hintley_name: 'wilhelm_huang_hintley_name',
        image: 'image url',
        poetry: 'poetry content',
        overview: 'overview',
        analysis: 'analysis',
        question: 'queston',
        line_13: 0,
        line_25: 0,
        line_46: 0,
        line_14: 0,
        line_36: 0,
        changing_lines: {
          line1: {
            img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
            line_13: [1, 1],
            line_14: [1, 1]
          },
          line2: {
            img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
            line_25: [1, 1]
          },
          line3: {
            img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
            line_13: [1, 1],
            line_36: [1, 1]
          },
          line4: {
            img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
            line_46: [1, 1],
            line_14: [1, 1]
          },
          line5: {
            img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
            line_25: [1, 1]
          },
          line6: {
            img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
            line_46: [1, 1],
            line_36: [1, 1]
          }
        }
      }
    };
    component.setProps({ hexagramArr: '7,9-7,9-7,9-7,9-7,9-7,9' });
    expect(renderer.create(component).toJSON()).toMatchSnapshot();
  });

  test('HexagramDetailModal has poetry font size snapshot', () => {
    const component = getShallowComponent();
    component.instance().hexagramsImgArrMap = {
      '7,9-7,9-7,9-7,9-7,9-7,9': {
        _id: 'id',
        number: 1,
        img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
        resonance_code_name: 'resonance_code_name',
        wilhelm_huang_hintley_name: 'wilhelm_huang_hintley_name',
        image: 'image url',
        poetry: 'poetry content',
        poetry_font_size: 18,
        overview: 'overview',
        analysis: 'analysis',
        question: 'queston',
        line_13: 0,
        line_25: 0,
        line_46: 0,
        line_14: 0,
        line_36: 0,
        changing_lines: {
          line1: {
            img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
            line_13: [1, 1],
            line_14: [1, 1]
          },
          line2: {
            img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
            line_25: [1, 1]
          },
          line3: {
            img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
            line_13: [1, 1],
            line_36: [1, 1]
          },
          line4: {
            img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
            line_46: [1, 1],
            line_14: [1, 1]
          },
          line5: {
            img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
            line_25: [1, 1]
          },
          line6: {
            img_arr: '7,9-7,9-7,9-7,9-7,9-7,9',
            line_46: [1, 1],
            line_36: [1, 1]
          }
        }
      }
    };
    component.setProps({ hexagramArr: '7,9-7,9-7,9-7,9-7,9-7,9' });
    expect(renderer.create(component).toJSON()).toMatchSnapshot();
  });*/
});
