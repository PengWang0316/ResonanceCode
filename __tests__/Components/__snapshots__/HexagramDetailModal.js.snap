// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HexagramDetailModal test HexagramDetailModal has poetry font size snapshot 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="hexagramDetailModalLabel"
  className="modal fade bd-example-modal-lg"
  id="hexagramDetailModal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog modal-lg"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="hexagramDetailModalLabel"
        >
           
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <ImageDescription
          imageInfo={
            Object {
              "_id": "id",
              "analysis": "analysis",
              "changing_lines": Object {
                "line1": Object {
                  "img_arr": "7,9-7,9-7,9-7,9-7,9-7,9",
                  "line_13": Array [
                    1,
                    1,
                  ],
                  "line_14": Array [
                    1,
                    1,
                  ],
                },
                "line2": Object {
                  "img_arr": "7,9-7,9-7,9-7,9-7,9-7,9",
                  "line_25": Array [
                    1,
                    1,
                  ],
                },
                "line3": Object {
                  "img_arr": "7,9-7,9-7,9-7,9-7,9-7,9",
                  "line_13": Array [
                    1,
                    1,
                  ],
                  "line_36": Array [
                    1,
                    1,
                  ],
                },
                "line4": Object {
                  "img_arr": "7,9-7,9-7,9-7,9-7,9-7,9",
                  "line_14": Array [
                    1,
                    1,
                  ],
                  "line_46": Array [
                    1,
                    1,
                  ],
                },
                "line5": Object {
                  "img_arr": "7,9-7,9-7,9-7,9-7,9-7,9",
                  "line_25": Array [
                    1,
                    1,
                  ],
                },
                "line6": Object {
                  "img_arr": "7,9-7,9-7,9-7,9-7,9-7,9",
                  "line_16": Array [
                    1,
                    1,
                  ],
                  "line_46": Array [
                    1,
                    1,
                  ],
                },
              },
              "image": "image url",
              "img_arr": "img_arr",
              "line_13": 0,
              "line_14": 0,
              "line_25": 0,
              "line_36": 0,
              "line_46": 0,
              "number": 1,
              "overview": "overview",
              "poetry": "poetry content",
              "poetry_font_size": 18,
              "question": "queston",
              "resonance_code_name": "resonance_code_name",
              "wilhelm_huang_hintley_name": "wilhelm_huang_hintley_name",
            }
          }
          imageNumber="img_arr"
          isBlack={true}
          isFirstImage={true}
        />
        <div
          className="mt-4 d-flex flex-wrap justify-content-center"
        >
          <BigramClockBig
            lineText="Resonance Line 2/5"
            position={0}
          />
          <BigramClockBig
            lineText="Agency/Process Line 1/4"
            position={0}
          />
          <BigramClockBig
            lineText="Agent/Identity Line 3/6"
            position={0}
          />
        </div>
        <div
          className="pl-4 mt-5 d-flex flex-wrap justify-content-around"
        >
          <BigramBlockBig
            isLine13={true}
            lineText="Particle Bigram: Line 1 and 3"
            position={0}
          />
          <BigramBlockBig
            lineText="Wave Bigram: Line 4 and 6"
            position={0}
          />
        </div>
        <div
          className="mt-4 imagePoetryDiv"
        >
          <img
            alt="hexagram"
            className="img-fluid float-sm-left image"
            src="image url"
          />
          <blockquote
            className="poetry"
            style={
              Object {
                "fontSize": "18px",
              }
            }
          >
            poetry content
          </blockquote>
        </div>
        <div
          className="mt-4 overViewDiv"
        >
          Overview:
        </div>
        <div
          className="preLineWhiteSpace"
        >
          overview
        </div>
        <div
          className="mt-4 font-weight-bold"
        >
          Question:
        </div>
        <div
          className="preLineWhiteSpace"
        >
          queston
        </div>
        <div
          className="mt-4 font-weight-bold"
        >
          Analysis:
        </div>
        <div
          className="mb-4 preLineWhiteSpace"
        >
          analysis
        </div>
        <ChangingLines
          hexagram={
            Object {
              "_id": "id",
              "analysis": "analysis",
              "changing_lines": Object {
                "line1": Object {
                  "img_arr": "7,9-7,9-7,9-7,9-7,9-7,9",
                  "line_13": Array [
                    1,
                    1,
                  ],
                  "line_14": Array [
                    1,
                    1,
                  ],
                },
                "line2": Object {
                  "img_arr": "7,9-7,9-7,9-7,9-7,9-7,9",
                  "line_25": Array [
                    1,
                    1,
                  ],
                },
                "line3": Object {
                  "img_arr": "7,9-7,9-7,9-7,9-7,9-7,9",
                  "line_13": Array [
                    1,
                    1,
                  ],
                  "line_36": Array [
                    1,
                    1,
                  ],
                },
                "line4": Object {
                  "img_arr": "7,9-7,9-7,9-7,9-7,9-7,9",
                  "line_14": Array [
                    1,
                    1,
                  ],
                  "line_46": Array [
                    1,
                    1,
                  ],
                },
                "line5": Object {
                  "img_arr": "7,9-7,9-7,9-7,9-7,9-7,9",
                  "line_25": Array [
                    1,
                    1,
                  ],
                },
                "line6": Object {
                  "img_arr": "7,9-7,9-7,9-7,9-7,9-7,9",
                  "line_16": Array [
                    1,
                    1,
                  ],
                  "line_46": Array [
                    1,
                    1,
                  ],
                },
              },
              "image": "image url",
              "img_arr": "img_arr",
              "line_13": 0,
              "line_14": 0,
              "line_25": 0,
              "line_36": 0,
              "line_46": 0,
              "number": 1,
              "overview": "overview",
              "poetry": "poetry content",
              "poetry_font_size": 18,
              "question": "queston",
              "resonance_code_name": "resonance_code_name",
              "wilhelm_huang_hintley_name": "wilhelm_huang_hintley_name",
            }
          }
          hexagramsImgArrMap={
            Object {
              "7,9-7,9-7,9-7,9-7,9-7,9": Object {
                "_id": "id",
                "number": "number",
                "resonance_code_name": "code name",
              },
            }
          }
        />
        <GroupHexagramTable
          handleHexagramClick={[MockFunction]}
          hexagramArray={
            Array [
              Object {
                "imageArr": undefined,
                "name": "Complementary",
                "number": undefined,
                "rcName": undefined,
              },
              Object {
                "imageArr": undefined,
                "name": "Reverse",
                "number": undefined,
                "rcName": undefined,
              },
              Object {
                "imageArr": undefined,
                "name": "Hidden Influence",
                "number": undefined,
                "rcName": undefined,
              },
            ]
          }
          tableTitle="Associated Hexagrams"
        />
        <GroupHexagramTable
          handleHexagramClick={[MockFunction]}
          hexagramArray={
            Array [
              Object {
                "imageArr": undefined,
                "name": "Potentiation",
                "number": undefined,
                "rcName": undefined,
              },
              Object {
                "imageArr": undefined,
                "name": "Growth",
                "number": undefined,
                "rcName": undefined,
              },
              Object {
                "imageArr": undefined,
                "name": "Maturation",
                "number": undefined,
                "rcName": undefined,
              },
              Object {
                "imageArr": undefined,
                "name": "Re-Sourcing",
                "number": undefined,
                "rcName": undefined,
              },
            ]
          }
          tableTitle="Resonance Quartet Hexagrams"
        />
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          type="button"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`HexagramDetailModal test HexagramDetailModal snapshot 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="hexagramDetailModalLabel"
  className="modal fade bd-example-modal-lg"
  id="hexagramDetailModal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog modal-lg"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="hexagramDetailModalLabel"
        >
           
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <ImageDescription
          imageInfo={
            Object {
              "_id": "id",
              "analysis": "analysis",
              "changing_lines": Object {
                "line1": Object {
                  "img_arr": "7,9-7,9-7,9-7,9-7,9-7,9",
                  "line_13": Array [
                    1,
                    1,
                  ],
                  "line_14": Array [
                    1,
                    1,
                  ],
                },
                "line2": Object {
                  "img_arr": "7,9-7,9-7,9-7,9-7,9-7,9",
                  "line_25": Array [
                    1,
                    1,
                  ],
                },
                "line3": Object {
                  "img_arr": "7,9-7,9-7,9-7,9-7,9-7,9",
                  "line_13": Array [
                    1,
                    1,
                  ],
                  "line_36": Array [
                    1,
                    1,
                  ],
                },
                "line4": Object {
                  "img_arr": "7,9-7,9-7,9-7,9-7,9-7,9",
                  "line_14": Array [
                    1,
                    1,
                  ],
                  "line_46": Array [
                    1,
                    1,
                  ],
                },
                "line5": Object {
                  "img_arr": "7,9-7,9-7,9-7,9-7,9-7,9",
                  "line_25": Array [
                    1,
                    1,
                  ],
                },
                "line6": Object {
                  "img_arr": "7,9-7,9-7,9-7,9-7,9-7,9",
                  "line_16": Array [
                    1,
                    1,
                  ],
                  "line_46": Array [
                    1,
                    1,
                  ],
                },
              },
              "image": "image url",
              "img_arr": "img_arr",
              "line_13": 0,
              "line_14": 0,
              "line_25": 0,
              "line_36": 0,
              "line_46": 0,
              "number": 1,
              "overview": "overview",
              "poetry": "poetry content",
              "question": "queston",
              "resonance_code_name": "resonance_code_name",
              "wilhelm_huang_hintley_name": "wilhelm_huang_hintley_name",
            }
          }
          imageNumber="img_arr"
          isBlack={true}
          isFirstImage={true}
        />
        <div
          className="mt-4 d-flex flex-wrap justify-content-center"
        >
          <BigramClockBig
            lineText="Resonance Line 2/5"
            position={0}
          />
          <BigramClockBig
            lineText="Agency/Process Line 1/4"
            position={0}
          />
          <BigramClockBig
            lineText="Agent/Identity Line 3/6"
            position={0}
          />
        </div>
        <div
          className="pl-4 mt-5 d-flex flex-wrap justify-content-around"
        >
          <BigramBlockBig
            isLine13={true}
            lineText="Particle Bigram: Line 1 and 3"
            position={0}
          />
          <BigramBlockBig
            lineText="Wave Bigram: Line 4 and 6"
            position={0}
          />
        </div>
        <div
          className="mt-4 imagePoetryDiv"
        >
          <img
            alt="hexagram"
            className="img-fluid float-sm-left image"
            src="image url"
          />
          <blockquote
            className="poetry"
            style={Object {}}
          >
            poetry content
          </blockquote>
        </div>
        <div
          className="mt-4 overViewDiv"
        >
          Overview:
        </div>
        <div
          className="preLineWhiteSpace"
        >
          overview
        </div>
        <div
          className="mt-4 font-weight-bold"
        >
          Question:
        </div>
        <div
          className="preLineWhiteSpace"
        >
          queston
        </div>
        <div
          className="mt-4 font-weight-bold"
        >
          Analysis:
        </div>
        <div
          className="mb-4 preLineWhiteSpace"
        >
          analysis
        </div>
        <ChangingLines
          hexagram={
            Object {
              "_id": "id",
              "analysis": "analysis",
              "changing_lines": Object {
                "line1": Object {
                  "img_arr": "7,9-7,9-7,9-7,9-7,9-7,9",
                  "line_13": Array [
                    1,
                    1,
                  ],
                  "line_14": Array [
                    1,
                    1,
                  ],
                },
                "line2": Object {
                  "img_arr": "7,9-7,9-7,9-7,9-7,9-7,9",
                  "line_25": Array [
                    1,
                    1,
                  ],
                },
                "line3": Object {
                  "img_arr": "7,9-7,9-7,9-7,9-7,9-7,9",
                  "line_13": Array [
                    1,
                    1,
                  ],
                  "line_36": Array [
                    1,
                    1,
                  ],
                },
                "line4": Object {
                  "img_arr": "7,9-7,9-7,9-7,9-7,9-7,9",
                  "line_14": Array [
                    1,
                    1,
                  ],
                  "line_46": Array [
                    1,
                    1,
                  ],
                },
                "line5": Object {
                  "img_arr": "7,9-7,9-7,9-7,9-7,9-7,9",
                  "line_25": Array [
                    1,
                    1,
                  ],
                },
                "line6": Object {
                  "img_arr": "7,9-7,9-7,9-7,9-7,9-7,9",
                  "line_16": Array [
                    1,
                    1,
                  ],
                  "line_46": Array [
                    1,
                    1,
                  ],
                },
              },
              "image": "image url",
              "img_arr": "img_arr",
              "line_13": 0,
              "line_14": 0,
              "line_25": 0,
              "line_36": 0,
              "line_46": 0,
              "number": 1,
              "overview": "overview",
              "poetry": "poetry content",
              "question": "queston",
              "resonance_code_name": "resonance_code_name",
              "wilhelm_huang_hintley_name": "wilhelm_huang_hintley_name",
            }
          }
          hexagramsImgArrMap={
            Object {
              "7,9-7,9-7,9-7,9-7,9-7,9": Object {
                "_id": "id",
                "number": "number",
                "resonance_code_name": "code name",
              },
            }
          }
        />
        <GroupHexagramTable
          handleHexagramClick={[MockFunction]}
          hexagramArray={
            Array [
              Object {
                "imageArr": undefined,
                "name": "Complementary",
                "number": undefined,
                "rcName": undefined,
              },
              Object {
                "imageArr": undefined,
                "name": "Reverse",
                "number": undefined,
                "rcName": undefined,
              },
              Object {
                "imageArr": undefined,
                "name": "Hidden Influence",
                "number": undefined,
                "rcName": undefined,
              },
            ]
          }
          tableTitle="Associated Hexagrams"
        />
        <GroupHexagramTable
          handleHexagramClick={[MockFunction]}
          hexagramArray={
            Array [
              Object {
                "imageArr": undefined,
                "name": "Potentiation",
                "number": undefined,
                "rcName": undefined,
              },
              Object {
                "imageArr": undefined,
                "name": "Growth",
                "number": undefined,
                "rcName": undefined,
              },
              Object {
                "imageArr": undefined,
                "name": "Maturation",
                "number": undefined,
                "rcName": undefined,
              },
              Object {
                "imageArr": undefined,
                "name": "Re-Sourcing",
                "number": undefined,
                "rcName": undefined,
              },
            ]
          }
          tableTitle="Resonance Quartet Hexagrams"
        />
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          type="button"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;
